core/src/main/java/org/apache/carbondata/core/datastore/page/LocalDictColumnPage.java

```diff
@@ -0,0 +1,319 @@
+
+  /**
+   * LOGGER
+   */
+  private static final LogService LOGGER =
+      LogServiceFactory.getLogService(LocalDictColumnPage.class.getName());
+
+  /**
+   * to maintain page level dictionary for column page
+   */
+  private PageLevelDictionary pageLevelDictionary;
+
+  /**
+   * to hold the actual data of the column
+   */
+  private ColumnPage actualDataColumnPage;
+
+  /**
+   * to hold the dictionary encoded column page
+   */
+  private ColumnPage encodedDataColumnPage;
+
+  /**
+   * to check if actual column page memory is already clear
+   */
+  private boolean isActualPageMemoryFreed;
+
+  /**
+   * Create a new column page with input data type and page size.
+   */
+  protected LocalDictColumnPage(ColumnPage actualDataColumnPage, ColumnPage encodedColumnpage,
+      LocalDictionaryGenerator localDictionaryGenerator) {
+    super(actualDataColumnPage.getColumnSpec(), actualDataColumnPage.getDataType(),
+        actualDataColumnPage.getPageSize());
+    if (!localDictionaryGenerator.isThresholdReached()) {
+      pageLevelDictionary = new PageLevelDictionary(localDictionaryGenerator,
+          actualDataColumnPage.getColumnSpec().getFieldName());
+      this.encodedDataColumnPage = encodedColumnpage;
+    }
+    this.actualDataColumnPage = actualDataColumnPage;
+  }
+
+  @Override public byte[][] getByteArrayPage() {
+    if (null != pageLevelDictionary) {
+      return encodedDataColumnPage.getByteArrayPage();
+    } else {
+      return actualDataColumnPage.getByteArrayPage();
+    }
+  }
+
+  /**
+   * Below method will be used to check whether page is local dictionary
+   * generated or not. This will be used for while enoding the the page
+   *
+   * @return
+   */
+  public boolean isLocalDictGeneratedPage() {
+    return null != pageLevelDictionary;
+  }
+
+  /**
+   * Below method will be used to add column data to page
+   *
+   * @param rowId row number
+   * @param bytes actual data
+   */
+  @Override public void putBytes(int rowId, byte[] bytes) {
+    if (null != pageLevelDictionary) {
+      try {
+        actualDataColumnPage.putBytes(rowId, bytes);
+        int dictionaryValue = pageLevelDictionary.getDictionaryValue(bytes);
+        encodedDataColumnPage.putBytes(rowId, ByteUtil.toBytes(dictionaryValue));
+      } catch (DictionaryThresholdReachedException e) {
+        LOGGER.error(e, "Local Dictionary threshold reached for the column: " + actualDataColumnPage
+            .getColumnSpec().getFieldName());
+        pageLevelDictionary = null;
+        encodedDataColumnPage.freeMemory();
+        encodedDataColumnPage = null;
+      }
+    } else {
+      actualDataColumnPage.putBytes(rowId, bytes);
+    }
+  }
+
+  public PageLevelDictionary getPageDictionary() {
+    return pageLevelDictionary;
+  }
+
+  @Override public void disableLocalDictEncoding() {
+    pageLevelDictionary = null;
+    freeEncodedColumnPage();
+  }
+
+  @Override public PageLevelDictionary getColumnPageDictionary() {
```

Comment:
duplicate method are present, remove one `getColumnPageDictionary` and `getPageDictionary`

Reply:
remove above method